source/common/filter/original_dst.cc

```diff
@@ -0,0 +1,36 @@
+#include "common/filter/original_dst.h"
```

Comment:
please put all listener filters in the `common/filter/listener` directory. It's bad that we don't already break things down like this. I would also use the `Envoy::Filter::Listener` namespace.

Reply:
Also, it somehow possible to split this PR into the core code first, and then follow up PRs that independently convert proxy proto and original DST to filters? That might not be easily possible but I figured I would ask. :)

Reply:
Will do the directory shuffle early next week. PR split is harder, as testing for correctness depends on the code being used. Also passing the coverage check might be problematic if the filters are split off.